pkg/getty/usr/bin/rungetty.sh

```diff
@@ -15,6 +15,14 @@ start_getty() {
 	term="linux"
 	[ "$speed" = "$1" ] && speed=115200
 
+	# did we already process this tty?
+	if $(echo "${PROCESSEDTTY}" | grep -q -w "$tty"); then
+		echo "getty: already processed tty for $tty, not starting twice" | tee /dev/console
+		return
+	fi
+	# now indicate that we are processing it
+	PROCESSEDTTY="${PROCESSEDTTY} ${tty}"
```

Comment:
One way to achieve a similar thing without quite so much shell gymnastics and grep etc would be `touch /var/lock/getty.${tty}` and `if [ -e /var/lock/getty.${tty} ]`.

Reply:
No preference. I am happy to leave it this way or change it. @ijc and @justincormack do you care?

Reply:
I think files are nicer than env vars. I would use `/tmp` though I think as its really only temporary to a short run of this program.

Reply:
It isn't really an env var, just a var in the context of the script (probably should not have named it in caps...).

Reply:
The more I think about it, more I think I would prefer to leave it as is. This isn't intended to survive process restarts or anything, as you said:

> its really only temporary to a short run of this program.

so why not just leave it as memory (var) inside program?

Reply:
The machinations to extract things from the var are the thing I don't like, rather than the fact of the var itself. Files would be cleaner in that regard.

Reply:
> machinations to extract things from the var